src/ophys_etl/modules/segmentation/graph_utils/community.py

```diff
@@ -0,0 +1,261 @@
+        thresh = vals.min() - (dip_fraction * vals.ptp())
+
+        if (vals.ptp() / vals.max()) > 0.05:
+            # terminates growth if the variation across the ROI gets too big
+            break
+
+        # get next layer of neighbors from this graph
+        neighbors = []
+        for n in last_sub:
+            neighbors.extend([i for i in graph.neighbors(n)
+                              if i not in last_sub])
+
+        # cycle through candidate neighbors
+        values = []
+        edges = []
+        for n in neighbors:
+            # for each neighbor candidate, connecting edges are
+            # from last_sub are:
+            node_edges = []
+            for sn in last_sub.nodes:
+                v = graph.get_edge_data(n, sn)
+                if v is not None:
+                    node_edges.append((n, sn, v))
+            node_edge_values = [i[2][attribute_name] for i in node_edges]
+            edges.append(node_edges)
+            # for this neighbor, the mean edge values connecting to last_sub
+            values.append(np.mean(node_edge_values))
+
+        if len(values) == 0:
+            return last_sub
+
+        # for the highest valued node, add all its edges
+        ind = np.argmax(values)
+        if values[ind] > thresh:
+            edges = edges[ind]
+            for edge in edges:
+                last_sub.add_edge(edge[0],
+                                  edge[1],
+                                  **edge[2])
+
+        n_nodes = len(last_sub)
+        if n_nodes > 200:
+            # in low-contrast regions, sometimes growth never stopped
+            return last_sub
+
+    print(f"grew subgraph from {n_start} to {n_nodes}")
+    return last_sub
+
+
+def iterative_detection(graph: Union[nx.Graph, Path],
+                        attribute_name: str,
+                        seed_quantile: int,
+                        n_node_thresh=20) -> Union[nx.Graph, Path]:
+    """idenitfy seeds, grow out ROIs, repeat
+
+    Parameters
+    ----------
+    graph: nx.Graph
+        the input graph, or a path to such
+    attribute_name: str
+        the name of the edge attribute
+    seed_quantile: float
+        establishes threshold for finding seeds
+    n_node_thresh: int
+        seeds smaller than this are disregarded
+
+    Returns
+    -------
+    graph: nx.Graph
+        a graph consisting of only identified ROIs, or a path to such
+
+    """
+    from_path = None
+    if isinstance(graph, Path):
+        from_path = Path(graph)
+        graph = nx.read_gpickle(graph)
+
+    collected = []
+    for jj in range(5):
+        subgraphs = seed_subgraphs_by_quantile(graph,
+                                               attribute_name=attribute_name,
+                                               seed_quantile=seed_quantile,
+                                               n_node_thresh=n_node_thresh)
+        if len(subgraphs) == 0:
+            break
+        expanded = []
+        for subgraph in subgraphs:
```

Comment:
If we go into production with this, I would like to consider introducing parallelization here, rather than at the global partition level.

Running a few test cases, I'm seeing behavior where some processes that were given easy subgraphs finish much earlier than process which were given difficult graphs.

I suppose the concern is that two processes may try to add the same node to different ROIs (since the processes won't have a globally updating `graph`). My reading of your algorithm, though, suggests this shouldn't happen, since `grow_subgraph` is designed not to cross any deep moats of low correlation.